server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdMonitor.java

```diff
@@ -196,30 +210,70 @@ public void onNewInfo(ClusterInfo info) {
         } else {
             listener.onResponse(null);
         }
+        // Get set of indices that are eligible to be automatically unblocked
+        // Only collect indices that are currently blocked
+        final String[] indices = state.routingTable().indicesRouting().keys().toArray(String.class);
+        Set<String> indicesToAutoRelease = Arrays.stream(indices)
+            .filter(index -> indicesToMarkIneligibleForAutoRelease.contains(index) == false)
+            .filter(index -> state.getBlocks().hasIndexBlock(index, IndexMetaData.INDEX_READ_ONLY_ALLOW_DELETE_BLOCK))
+            .collect(Collectors.toCollection(HashSet::new));
+
+        if (indicesToAutoRelease.isEmpty() == false) {
+            if (diskThresholdSettings.isAutoReleaseIndexEnabled()) {
+                logger.info("Releasing read-only allow delete block on indices: [{}]", indicesToAutoRelease);
+                updateIndicesReadOnly(indicesToAutoRelease, listener, false);
+            } else {
+                deprecationLogger.deprecated("[{}] will be removed in 8.0.0", DiskThresholdSettings.AUTO_RELEASE_INDEX_ENABLED_KEY);
```

Comment:
Suggest using a version constant to ensure that we clean this up in a future version:

```suggestion
                deprecationLogger.deprecated("[{}] will be removed in version {}", DiskThresholdSettings.AUTO_RELEASE_INDEX_ENABLED_KEY,
                    Version.V_7_4_0.major + 1);
```

Also, please emit a deprecation warning in the constructor if `diskThresholdSettings.isAutoReleaseIndexEnabled() == false`, otherwise we won't emit a warning on clusters that never run out of disk space.